lemur/plugins/lemur_gcp/endpoints.py

```diff
@@ -150,25 +150,25 @@ def update_target_proxy_sni_certs(project_id, credentials, endpoint, old_cert, n
     if kind not in ("targethttpsproxy", "targetsslproxy") or endpoint.registry_type != "gcp":
         raise NotImplementedError()
     current_app.logger.info(f"Rotating SNI cert for endpoint {endpoint.name}")
-    old_cert_name = certificates.get_name(old_cert.body)
-    old_cert_self_link = certificates.get_self_link(project_id, old_cert_name)
     new_cert_name = certificates.get_name(new_cert.body)
     new_cert_self_link = certificates.get_self_link(project_id, new_cert_name)
     if kind == "targethttpsproxy":
         client = target_https_proxies.TargetHttpsProxiesClient(credentials=credentials)
         proxy = client.get(project=project_id, target_https_proxy=endpoint.name)
-        if old_cert_self_link not in proxy.ssl_certificates:
-            current_app.logger.warning(f"Old cert {old_cert} found by Lemur but not in GCP - proceeding with "
+        cert_to_delete = certificates.find_cert(project_id, credentials, old_cert.body, proxy.ssl_certificates)
+        if cert_to_delete not in proxy.ssl_certificates:
+            current_app.logger.warning(f"Old cert {old_cert} found by Lemur but not in endpoint - proceeding with "
                                        "rotation anyway as detaching a non-existent cert is a no-op.")
-        certs = certificates.calc_diff(proxy.ssl_certificates, new_cert_self_link, old_cert_self_link)
+        certs = certificates.calc_diff(proxy.ssl_certificates, new_cert_self_link, cert_to_delete)
         set_target_https_proxy_certs(project_id, client, endpoint, certs, proxy.ssl_certificates)
     elif kind == "targetsslproxy":
         client = target_ssl_proxies.TargetSslProxiesClient(credentials=credentials)
         proxy = client.get(project=project_id, target_ssl_proxy=endpoint.name)
-        if old_cert_self_link not in proxy.ssl_certificates:
-            current_app.logger.warning(f"Old cert {old_cert} found by Lemur but not in GCP - proceeding with "
+        cert_to_delete = certificates.find_cert(project_id, credentials, old_cert.body, proxy.ssl_certificates)
+        if cert_to_delete not in proxy.ssl_certificates:
```

Comment:
maybe just check `if not cert_to_delete` because otherwise reading this could imply that there may be a cert to delete that's otherwise not attached to the proxy?